PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/mysql/AddMySQLDependencyAction.java

```diff
@@ -0,0 +1,175 @@
+import com.intellij.openapi.module.ModuleTypeId;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.project.Project;
+import com.microsoft.azure.common.exceptions.AzureExecutionException;
+import com.microsoft.azure.toolkit.lib.common.utils.MavenDependencyUtils;
+import com.microsoft.azuretools.ijidea.utility.AzureDependencyActionBase;
+import com.microsoft.azuretools.telemetry.TelemetryConstants;
+import com.microsoft.azuretools.telemetrywrapper.Operation;
+import com.microsoft.intellij.maven.DependencyArtifact;
+import com.microsoft.intellij.maven.SpringCloudDependencyManager;
+import com.microsoft.intellij.util.MavenUtils;
+import com.microsoft.intellij.util.PluginUtil;
+import com.microsoft.tooling.msservices.components.DefaultLoader;
+import org.apache.commons.lang3.StringUtils;
+import org.dom4j.DocumentException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.idea.maven.project.MavenProject;
+import org.jetbrains.idea.maven.project.MavenProjectsManager;
+import org.jetbrains.idea.maven.utils.MavenProcessCanceledException;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class AddMySQLDependencyAction extends AzureDependencyActionBase {
+    private static final String GROUP_ID = "mysql";
+    private static final String ARTIFACT_ID = "mysql-connector-java";
+
+    @Override
+    public boolean onActionPerformed(@NotNull AnActionEvent event, @Nullable Operation operation) {
+        final Module module = event.getData(LangDataKeys.MODULE);
+        final Project project = module.getProject();
+        final MavenProjectsManager projectsManager = MavenProjectsManager.getInstance(project);
+        final MavenProject mavenProject = projectsManager.findProject(module);
+        if (mavenProject == null) {
+            PluginUtil.showErrorNotificationProject(project, "Error",
+                                                    String.format("Project '%s' is not a maven project.",
+                                                                  project.getName()));
+            return true;
+        }
+
+        DefaultLoader.getIdeHelper().runInBackground(project,
+                                                     "Add MySQL dependency",
+                                                     false,
+                                                     true,
+                                                     "Add MySQL dependency", () -> {
+                ProgressIndicator progressIndicator = ProgressManager.getInstance().getProgressIndicator();
+                progressIndicator.setText("Syncing maven project " + project.getName());
+                if (!syncMavenDependency(projectsManager, project, mavenProject)) {
+                    return;
```

Comment:
do we need to prompt any messages here?

Reply:
syncMavenDependency will show error when syncing error